sdk/core/src/policies/mod.rs

```diff
@@ -0,0 +1,26 @@
+mod retry_policies;
+mod transport;
+
+use crate::{Context, Request, Response};
+pub use retry_policies::*;
+use std::error::Error;
+use std::sync::Arc;
+pub use transport::*;
+
+pub type PolicyResult<T> = Result<T, Box<dyn Error + Send + Sync>>;
+
+/// A pipeline policy.
+///
+/// Policies are expected to modify the request and then call the following policy.
+/// Policies can then inspect the response, potentially signaling failure.
+/// The only runtime enforced check is that the last policy must be a Transport policy. It's up to
+/// the implementer to call the following policy.
+#[async_trait::async_trait]
+pub trait Policy: Send + Sync + std::fmt::Debug {
+    async fn send(
```

Comment:
Policies must receive both the request and response to work. See https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/HttpPipelinePolicy.cs. In .NET, we wrap those in a `Message`. This is critical because some policies act on data from the response itself, like challenge-based auth or the `RetryPolicy` (e.g. `Retry-After` header, which could be either an int (seconds) or string (ISO8601 datetime).

Reply:
The policies do receive both. They receive the request from the caller and then they call the next policy which gives them a response.